query/expression.go

```diff
@@ -34,7 +34,8 @@ type EvaluationContext struct {
 	API          api.API          // Api to obtain metadata from
 	Timerange    api.Timerange    // Timerange to fetch data from
 	SampleMethod api.SampleMethod // SampleMethod to use when up/downsampling to match the requested resolution
-	Predicate    api.Predicate
+	Predicate    api.Predicate    // Predicate to apply to TagSets prior to fetching
+	FetchLimit   *int             // A limit on the number of fetches which may be performed
```

Comment:
@achow oh, i didn't read the last comment.

i think in sort term, this being pointer is a better solution because there are few stateful and stateless values in this object. (timerange for timeshift for example). being able to safely copy and make modifications to it is a good property to have.

if we make `EvaluationContext` as a pointer (and keep the limit as value), then we have to re-evaluate how timeshifting needs to be done - because we cannot simply modify it, since it's a global object.

in future, i feel like it's worthwhile to categorize fields that can freely be changed per sub-expression node and data that must be shared between expressions.